sdk/communication/Azure.Communication.Sms/src/autorest.md

```diff
@@ -12,4 +12,5 @@ require:
     -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/896d05e37dbb00712726620b8d679cc3c3be09fb/specification/communication/data-plane/Sms/readme.md
 payload-flattening-threshold: 10
 generation1-convenience-client: true
+model-factory-for-hlc: true
```

Comment:
nit: seems we do this quite a bit would it be better to make this an "opt out"?

Reply:
Well this is already an "opt out" for our new model factories. This is to restore the old behaviour of the model factory in hlc.
FYI the old model factory in hlc works like this:
it only has the models that have at least one readonly property.
it will always put discriminator in the method.

I really do not want to add customization code for them therefore I added this configuration for them so that these libraries could keep their old code. When they want to adopt our new style of model factories, they could just remove this config

Reply:
@ArcturusZhang can we explore whether or not filtering models by ones with readonly properties would work for us as well?

I guess one issue would be if something goes from readonly to not we wouldn't want the factory to disappear?

Reply:
I have changed this. Now we always only generate models with at least one readonly properties.